lib/rules/no-unused-vars.js

```diff
@@ -832,6 +836,588 @@ module.exports = {
+
+            /**
+             * Check if an array has a single element with null as other element.
+             * @param {ASTNode} node ArrayPattern node
+             * @returns {boolean} true if array has single element with other null elements
+             */
+            function hasSingleElement(node) {
+                return node.elements.filter(e => e !== null).length === 1;
+            }
+
+            /**
+             * check whether import specifier has an import of particular type
+             * @param {ASTNode} node ImportDeclaration node
+             * @param {string} type type of import to check
+             * @returns {boolean} true if import specifier has import of specified type
+             */
+            function hasImportOfCertainType(node, type) {
+                return node.specifiers.some(e => e.type === type);
+            }
+
+            /**
+             * give fixes for unused variables in function parameters
+             * @param {ASTNode} node node to check
+             * @returns {Object} fixer object
+             */
+            function fixFunctionParameters(node) {
+                const parentNode = node.parent;
+
+                if (isFunction(parentNode)) {
+
+                    // remove unused function parameter if there is only a single parameter
+                    if (parentNode.params.length === 1) {
+                        return fixer.removeRange(node.range);
+                    }
+
+                    // remove first unused function parameter when there are multiple parameters
+                    if (getTokenBeforeValue(node) === "(" && getTokenAfterValue(node) === ",") {
+                        return fixer.removeRange([node.range[0], getNextTokenEnd(node)]);
+                    }
+
+                    // remove unused function parameters except first one when there are multiple parameters
+                    return fixer.removeRange([getPreviousTokenStart(node), node.range[1]]);
+                }
+
+                return null;
+            }
+
+            /**
+             * Gets fixes in variable declarations and function parameters
+             * @param {ASTNode} node parent node of identifier
+             * @returns {Object} fixer object
+             */
+            function fixVariables(node) {
+                const parentNode = node.parent;
+
+                // remove unused declared variables such as var a = b; or var a = b, c;
+                if (parentNode.type === "VariableDeclarator") {
+
+                    // remove unused declared variable with single declaration such as 'var a = b;'
+                    if (parentNode.parent.declarations.length === 1) {
+                        return fixer.removeRange(parentNode.parent.range);
+                    }
+
+                    // remove unused declared variable with multiple declaration except first one such as 'var a = b, c = d;'
+                    if (getTokenBeforeValue(parentNode) === ",") {
```

Comment:
Here in this suggestion these conditions will only be applied when there are multiple declarations `(var a, b;)` as they are under this condition:
```js
if (getTokenBeforeValue(parentNode) === ",") {// code}
```

i guess intend here is to prevent suggestions when there is only one declaration, could you please clarify that?

Reply:
You are correct! These checks should be added to the previous if-block, where we test that there is exactly 1 declaration. I've added this suggestion at the correct line now. Sorry for the confusion.